security/scripts/precommit-fast-tier.sh

```
@@ -0,0 +1,39 @@
+ROOT="$(git rev-parse --show-toplevel 2>/dev/null)"
+[[ -z "$ROOT" ]] && exit 0
+cd "$ROOT" || exit 0
+
+STAGED_GO=$(git diff --cached --name-only --diff-filter=ACM | grep -E '\.go$' || true)
+STAGED_SENSITIVE=$(git diff --cached --name-only --diff-filter=ACM | grep -E '(internal/session/raw_writer\.go|internal/session/redactor|internal/session/gitleaks_|cmd/ox/prepush_scan\.go|internal/auth/|internal/mcp/|internal/exec/|internal/git/|cmd/ox/upgrade\.go|internal/upgrade/|^go\.sum$|^go\.mod$)' || true)
+
+if [[ -n "$STAGED_GO" ]] && command -v govulncheck >/dev/null 2>&1; then
+  echo "[SEC_PRECOMMIT] govulncheck on touched Go packages..."
+  PKGS=$(echo "$STAGED_GO" | xargs -n1 dirname 2>/dev/null | sort -u | sed 's|^|./|')
+  if [[ -n "$PKGS" ]]; then
+    govulncheck $PKGS 2>&1 | head -50 || echo "[SEC_PRECOMMIT] govulncheck warnings (advisory, not blocking)"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)